### PR TITLE
Set informatieobjecttype

### PR DESCRIPTION
**Pull request by bbrands02**

### CODE REVIEW

**Review by smisidjan — APPROVED** [review]

File: src/Service/WPZaakService.php [hunk 1]
```diff
@@ -132,6 +132,36 @@ private function getRSIN(array $zaak): ?string
...
+                $synchronization = $this->entityManager->getRepository('App:Synchronizations')->findOneBy(['sourceId' => $id]);
+                if ($synchronization !== null) {
+                    return $synchronization->getObject()->getId()->toString();
+                }
+            }
+        }
+
+        return null;
+
+    }//end getInformatieObjectTypeId()
```
Comment (rjzondervan) [inline src/Service/WPZaakService.php]: I have to check, but I thought we had decided sometime that sourceId would be the url of the object in most cases, which would mean we would not need all of this.

However, more than fine for the moment.

**Review by rjzondervan — APPROVED** [review]

File: src/Service/WPZaakService.php [hunk 2]
```diff
@@ -218,7 +248,7 @@ public function saveWaardepapierInDRC(string $data, ObjectEntity $zaakObject): v
 
         $informationArray = [
             'inhoud'                       => base64_encode($data),
-            'informatieobjecttype'         => $this->resourceService->getObject($this->configuration['informatieobjecttype']),
+            'informatieobjecttype'         => $this->getInformatieObjectTypeId(),
```
Comment (rjzondervan) [inline src/Service/WPZaakService.php]: This should always be a URL

**Review by rjzondervan — CHANGES_REQUESTED** [review]

File: src/Service/WPZaakService.php (hunk 1 above)
Comment (WilcoLouwerse) [reply src/Service/WPZaakService.php]: Just putting it out there, that this is not how the use of synchronizations was/is intended. If it is needed as a work around we can set sourceId to a URL.
But we should use Source->location + Synchronization->endpoint + Synchronization->sourceId as combination to do API calls. So SourceId should be in most cases the id on the source

File: src/Service/WPZaakService.php [hunk 3]
```diff
@@ -132,6 +132,36 @@ private function getRSIN(array $zaak): ?string
...
+     * Finds a informatieobjectttype url from a object url in the action configuration.
+     *
+     * @return string|null
+     */
+    private function getInformatieObjectTypeUrl()
+    {
+        $informatieobjecttype = ($this->configuration['informatieobjecttype'] ?? '');
+        if (empty($informatieobjecttype) === false) {
+            $uuidPattern = '/[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}/i';
+            if (preg_match($uuidPattern, $informatieobjecttype, $matches)) {
```
Comment (WilcoLouwerse) [inline src/Service/WPZaakService.php:145]: we can't just use ramsey uuid check here?

**Review by WilcoLouwerse — APPROVED** [review]

File: src/Service/WPZaakService.php (hunk 1 above)
Comment (rjzondervan) [reply src/Service/WPZaakService.php]: Usually I would agree, however, by only storing the id we have the risk of multiple objects with the same source id, as not everybody uses uuid as id (a simple int is still very, very, very usual). The only way to be almost certain (I have seen APIs with multiple _different_ objects with the same id in the same endpoint) you have a unique object on the source is by storing the full url.

**Pull request merged**